src/plugins/mf_priority.cpp

```diff
@@ -101,28 +102,43 @@ static void rec_update_cb (flux_t *h,
                            const flux_msg_t *msg,
                            void *arg)
 {
-    char *uid, *fshare, *bank, *default_bank, *max_jobs;
-
-    if (flux_request_unpack (msg, NULL, "{s:s, s:s, s:s, s:s, s:s}",
-                             "userid", &uid,
-                             "bank", &bank,
-                             "default_bank", &default_bank,
-                             "fairshare", &fshare,
-                             "max_jobs", &max_jobs) < 0) {
+    char *uid, *fshare, *bank, *def_bank, *max_jobs;
+    json_t *data, *jtemp = NULL;
+    json_error_t error;
+    int num_data = 0;
+
+    if (flux_request_unpack (msg, NULL, "{s:o}", "data", &data) < 0) {
         flux_log_error (h, "failed to unpack custom_priority.trigger msg");
         goto error;
     }
 
     if (flux_respond (h, msg, NULL) < 0)
         flux_log_error (h, "flux_respond");
 
-    struct bank_info *b;
-    b = &users[std::atoi(uid)][bank];
+    if (!data || !json_is_array (data)) {
+        flux_log (h, LOG_ERR, "mf_priority: invalid bulk_update payload");
+        goto error;
+    }
+    num_data = json_array_size (data);
 
-    b->fairshare = std::atof (fshare);
-    b->max_jobs = std::atoi (max_jobs);
+    for (int i = 0; i < num_data; i++) {
+       json_t *el = json_array_get(data, i);
 
-    users_def_bank[std::atoi (uid)] = default_bank;
+       if ((json_unpack_ex (el, &error, 0, "{s:s}", "userid", &uid) ||
+            json_unpack_ex (el, &error, 0, "{s:s}", "bank", &bank) ||
+            json_unpack_ex (el, &error, 0, "{s:s}", "def_bank", &def_bank) ||
+            json_unpack_ex (el, &error, 0, "{s:s}", "fairshare", &fshare) ||
+            json_unpack_ex (el, &error, 0, "{s:s}", "max_jobs", &max_jobs)) < 0)
```

Comment:
You can combine these 5 `json_unpack_ex()` calls into one:
```c
 if (json_unpack_ex (el, &error, 0,
                     "{s:s s:s s:s s:s s:s}",
                     "userid", &uid,
                     "bank", &bank,
                     "def_bank", &def_bank,
                     "fairshare", &fshare,
                     "max_jobs", &max_jobs") < 0)
                               
```

Reply:
Also, instead of encoding everything as a string, a possible improvement would be to encode `fairshare`  as a JSON real and `max_jobs` as an int, and unpack directly to these values, e.g.

```c
if (json_unpack_ex (el, &error, 0,
                     "{s:i s:s s:s s:F s:i}",
                     "userid", &uid,
                     "bank", &bank,
                     "def_bank", &def_bank,
                     "fairshare", &fshare,
                     "max_jobs", &max_jobs") < 0)
```

Then you would not have to convert all values to `str` in the bulk update script that sends this data. (One of the nice thing about JSON is that you can encode these different types, so might as well use it)

Reply:
Ah, I think I understand what you are suggesting. Thanks. I just pushed up a change that combines all of the unpacks into one as well as encoded the necessary data into their resepective types (`uid` and `max_jobs` into `int` and `fshare` as a JSON real). Like you mentioned above, this included minor changes and updates to the bulk update script and the priority sharness tests, which I've squashed.